crates/proto/src/error.rs

```diff
@@ -345,6 +345,29 @@ pub enum ProtoErrorKind {
     NativeCerts,
 }
 
+/// Data needed to process a SOA-record-based referral.
+#[derive(Clone, Debug)]
+pub struct ForwardData {
+    /// Name
+    pub name: Name,
+    /// No records found?
+    pub is_no_records_found: bool,
+    /// IS nx domain?
+    pub is_nx_domain: bool,
+}
+
+impl ForwardData {
+    /// are there records?
+    pub fn is_no_records_found(&self) -> bool {
+        self.is_no_records_found
+    }
+
+    /// is this nxdomain?
+    pub fn is_nx_domain(&self) -> bool {
+        self.is_nx_domain
+    }
```

Comment:
No point in these if the fields are `pub`.

Reply:
These will probably stay pub in the final version of this, and the fields will either go away or be private.

Reply:
Hmm, for this type I kinda like the `pub` fields? Is `no_records_found: true, nx_domain: true` an invalid state? What would be the reason to make these private?

Reply:
For now, consistency with the other error types.  is_nx_domain() and is_no_records_found() are in the Error implementations in the recursor, resolver, and server.